File: Source Code/VB/Chapter03/SimpleLibrary/Fibonacci.vb
Public Class Fibonacci
    Public Shared Function Calculate(ByVal x As Integer) As Integer
        If x < 0 Then
            Throw New ArgumentException("x must be greater than or equal to zero")
        End If
        If x < 2 Then
            Return x
        End If
        Return Calculate(x - 1) + Calculate(x - 2)
    End Function
End Class

File: Source Code/CSharp/Chapter03/SimpleLibrary/Fibonacci.cs
using System;

namespace SimpleLibrary
{
    public class Fibonacci
    {
        public static int Calculate(int x)
        {
            if (x < 0)
                throw new ArgumentException("x must be greater than or equal to zero");
            return (x > 1) ? Calculate(x - 1) + Calculate(x - 2) : x;
        }
    }
}
